executor/api/rest/server.go

```diff
@@ -142,8 +142,9 @@ func (r *SeldonRestApi) Initialise() {
 func puidHeader(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		if puid := r.Header.Get(payload.SeldonPUIDHeader); puid == "" {
-			r.Header.Set(payload.SeldonPUIDHeader, guuid.New().String())
```

Comment:
Are you sure this is correct as this adds a header to the request if it did not exist.
Don't we need both?

Reply:
Thanks for the check, I've added now the logic to set both the request and response, as well as a test to confirm that it's set, please have a look